gtsfm/common/gtsfm_data.py

```diff
@@ -27,15 +28,30 @@ class GtsfmData:
     The situation of non-contiguous cameras can exists because of failures in front-end.
     """
 
-    def __init__(self, number_images: int) -> None:
+    def __init__(
+        self,
+        number_images: int,
+        cameras: Optional[Dict[int, PinholeCameraCal3Bundler]] = None,
+        tracks: Optional[List[SfmTrack]] = None,
+        scene_mesh: Optional[Trimesh] = None,
+    ) -> None:
         """Initializes the class.
 
         Args:
             number_images: number of images/cameras in the scene.
         """
-        self._cameras: Dict[int, PinholeCameraCal3Bundler] = {}
-        self._tracks: List[SfmTrack] = []
         self._number_images = number_images
+        self._cameras = cameras if cameras else {}
+        self._tracks = tracks if tracks else []
+        # self._cameras = cameras  # this caused an error for some reason??
+        # self._tracks = tracks
+        self.scene_mesh = scene_mesh
+
+    def get_two_view_data(self, i1: int, i2: int) -> "GtsfmData":
+        """Collects GtsfmData for a sinbgle image pair."""
+        # TODO (travisdriver): also collect tracks if available.
+        cameras_pair = {0: self.get_camera(i1), 1: self.get_camera(i2)}
+        return GtsfmData(2, cameras_pair, None, self.scene_mesh)
```

Comment:
Whats the reason for introducing this API?

Also nit: typo in the docstring.

Reply:
So that we don't need to pass the entire GtsfmData object to the TwoViewEstimator, as it could contain thousands of cameras. Also, it conforms with the previous API.

Reply:
fixed typo